rdrf/rdrf/reports/generator.py

```diff
@@ -0,0 +1,707 @@
+        elif db == "default":
+            self.reporting_engine = self.default_engine
+        elif db == "reporting":
+            self.reporting_engine = self._create_engine("reporting")
+        else:
+            raise Exception(
+                "Unknown db: %s. Should be one of clinical | default" % db)
+
+    def mkcol(self, triple, form_model=None, section_model=None, cde_model=None, field=None):
+        column_name = triple[0]
+        column_datatype = triple[1]
+        is_nullable = triple[2]
+        column = alc.Column(column_name, column_datatype, nullable=is_nullable)
+        datasource = DataSource(self.registry_model,
+                                column,
+                                form_model=form_model,
+                                section_model=section_model,
+                                cde_model=cde_model,
+                                field=field)
+
+        self.column_map[column] = datasource
+        return column
+
+    def _create_engine(self, db_name="default"):
+        # we should probably add a reporting db ...
+        return create_engine(pg_uri(settings.DATABASES[db_name]))
+
+    def _get_table_for_model(self, model, db_name="default"):
+
+        if db_name == "default":
+            engine = self.default_engine
+        else:
+            engine = self.engine
+
+        table = alc.Table(model._meta.db_table,
+                          MetaData(engine), autoload=True)
+        return table
+
+    def _copy_table_data(self, src_engine, dest_engine, table):
+        dest_engine.execute("DROP TABLE IF EXISTS %s CASCADE" % table.name)
+        table.create(dest_engine)
+        rows = src_engine.execute(table.select()).fetchall()
+
+        with dest_engine.begin() as con:
+            for row in rows:
+                con.execute(table.insert().values(**row))
+
+    def _get_sql_alchemy_type(self, db_type):
+        return alc.String
+
+    def clear(self):
+        # drop tables etc
+        for table in self.table_list:
+            self._drop_table(table)
+
+    def _create_demographic_tables(self):
+        from registry.patients.models import PatientAddress, AddressType, State, NextOfKinRelationship
+        from registry.patients.models import ConsentValue
+        from rdrf.models import ConsentQuestion, ConsentSection, Registry, RegistryForm
+        from rdrf.models import Section, CommonDataElement, CDEPermittedValueGroup, CDEPermittedValue
+        from explorer.models import Query
+
+        starting_models = [ContentType, Registry, Group, State, AddressType, NextOfKinRelationship,
+                           PatientAddress, Query, Section, ConsentSection, ConsentQuestion,ConsentValue,
+                           RegistryForm, CDEPermittedValue,
+                           CDEPermittedValueGroup, CommonDataElement, Patient]
+
+        if self.reporting_engine is self.default_engine:
+            raise Exception("reporting db = default!")
+
+        models = []
```

Comment:
make this a `dict` of models keyed by `model.name`? Would simplify the add code below...